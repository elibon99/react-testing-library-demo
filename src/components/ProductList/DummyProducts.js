import Macbook from "../../assets/images/Macbook.jpg"
import Iphone from "../../assets/images/Iphone.jpg"
import AirPods from "../../assets/images/AirPods.webp"

export const DummyProducts = [
    {
        id: 1,
        title: "Macbook",
        price: 16000,
        image: Macbook
    },
    {
        id: 2,
        title: "Iphone",
        price: 7000,
        image: Iphone
    },
    {
        id: 3,
        title: "AirPods",
        price: 1500,
        image: AirPods
    }
]