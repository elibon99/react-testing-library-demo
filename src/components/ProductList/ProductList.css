
.product-list-container {
    display: flex;
    flex-direction: column;
    background: white;
    border-radius: 8px;
    width: 65%;
    padding: 30px 30px;
}

.product-list {
    display: flex;
    justify-content: space-between;
    margin-top: 20px;
    align-items: stretch;
}

.product-list-header {
    margin: 0 0;
}