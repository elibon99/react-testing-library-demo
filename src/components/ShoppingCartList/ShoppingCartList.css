
.shopping-cart-list-container {
    display: flex;
    flex-direction: column;
    background: white;
    width: 20%;
    padding: 30px 30px;
    border-radius: 8px;
}

.shopping-cart-list-inner-container {
    display: flex;
    flex-direction: column;
    margin-top: 20px;
}

.shopping-cart-list-header {
    margin: 0 0;
}