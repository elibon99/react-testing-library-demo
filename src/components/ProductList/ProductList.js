import React from 'react'
import "./ProductList.css"
import {DummyProducts} from "./DummyProducts";
import ProductItem from "../ProductItem/ProductItem";

const ProductList = ({products, setProducts}) => {
    const addProductToCart = (id) => {
        let product = DummyProducts.find((product) => product.id === id)
        setProducts(prev => [...prev, product])
    }

    return (
        <div className="product-list-container">
            <h3 className="product-list-header">Products</h3>
            <div className="product-list" data-testid={"product-list"}>
                {DummyProducts.map((product, index) => {
                    return(
                        <ProductItem key={index} product={product} products={products} addProductToCart={addProductToCart}/>
                    )
                })}
            </div>
        </div>
    )
}

export default ProductList