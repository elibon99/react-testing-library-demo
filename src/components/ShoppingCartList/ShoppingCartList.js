import React from 'react'
import "./ShoppingCartList.css"
import ShoppingCartItem from "../ShoppingCartItem/ShoppingCartItem";
import ShoppingStatus from "../ShoppingStatus/ShoppingStatus";

const ShoppingCartList = ({products, setProducts}) => {
    const removeProduct = (id) => {
        let updatedProducts = products.filter((product) => product.id !== id)
        setProducts(updatedProducts)
    }

    return (
        <div className="shopping-cart-list-container">
            <h3 className="shopping-cart-list-header">Cart Items</h3>
            <div className="shopping-cart-list-inner-container" data-testid={"shopping-cart-list"}>
                {products && products.map((product, index) => {
                    return(
                        <ShoppingCartItem product={product} removeProduct={removeProduct} key={index}/>
                    )
                })}
            </div>
            <ShoppingStatus products={products}/>
        </div>
    )
}

export default ShoppingCartList
