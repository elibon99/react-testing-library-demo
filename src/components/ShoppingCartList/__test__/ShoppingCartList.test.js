import {render, screen} from "@testing-library/react";
import ShoppingCartList from "../ShoppingCartList";
import {DummyProducts} from "../../ProductList/DummyProducts";

const ShoppingCart = DummyProducts.filter((product) => product.title === "Macbook")

describe('<ShoppingCartList />', () => {

    /**
     * Checks that ShoppingCartList is rendered correctly by comparing with snapshot
     */
    it('renders correctly', async () => {
        const {asFragment} = render(
            <ShoppingCartList
                products={ShoppingCart}
                setProducts={() => {}}
            />
        )
        expect(asFragment()).toMatchSnapshot()
    })
})