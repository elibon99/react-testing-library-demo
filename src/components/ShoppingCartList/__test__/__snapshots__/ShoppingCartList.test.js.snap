// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShoppingCartList /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="shopping-cart-list-container"
  >
    <h3
      class="shopping-cart-list-header"
    >
      Cart Items
    </h3>
    <div
      class="shopping-cart-list-inner-container"
      data-testid="shopping-cart-list"
    >
      <div
        class="shopping-cart-item-container"
        data-testid="shopping-cart-item"
      >
        <h4
          class="shopping-cart-item-title"
        >
          Macbook
        </h4>
        <div
          class="shopping-cart-item-remove"
          data-testid="remove-product-button"
        >
          <svg
            aria-hidden="true"
            height="28"
            preserveAspectRatio="xMidYMid meet"
            role="img"
            viewBox="0 0 64 64"
            width="28"
            xlink="http://www.w3.org/1999/xlink"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M62 52c0 5.5-4.5 10-10 10H12C6.5 62 2 57.5 2 52V12C2 6.5 6.5 2 12 2h40c5.5 0 10 4.5 10 10v40z"
              fill="#ff5a79"
            />
            <path
              d="M50 21.2L42.8 14L32 24.8L21.2 14L14 21.2L24.8 32L14 42.8l7.2 7.2L32 39.2L42.8 50l7.2-7.2L39.2 32z"
              fill="#fff"
            />
          </svg>
        </div>
        <h4
          class="shopping-cart-item-price"
        >
          16000 SEK
        </h4>
      </div>
    </div>
    <div
      class="shopping-status-container"
    >
      <h3
        class="shopping-status-title"
      >
        Total items: 1
      </h3>
      <h3
        class="shopping-status-title"
      >
        Total price: 16000
      </h3>
    </div>
  </div>
</DocumentFragment>
`;
